site/content/docs/main/guides/prometheus.md

```diff
@@ -28,67 +28,60 @@ Contour exposes a Prometheus-compatible `/metrics` endpoint that defaults to lis
 
 {{% metrics-table %}}
 
-## Sample Deployment
+## Deploy Sample Monitoring Stack
 
-In the `/examples` directory there are example deployment files that can be used to spin up an example environment.
-All deployments there are configured with annotations for prometheus to scrape by default, so it should be possible to utilize any of them with the following quickstart example instructions.
+Follow the instructions [here][0] to install a monitoring stack to your cluster using the [kube-prometheus][1] project sample manifests.
+These instructions install the [Prometheus Operator][2], a [Prometheus][3] instance, a [Grafana][4] `Deployment`, and other components.
+Note that this is a quickstart installation, see documentation [here][5] for more details on customizing the installation for production usage.
 
-### Deploy Prometheus
-
-A sample deployment of Prometheus and Alertmanager is provided that uses temporary storage. This deployment can be used for testing and development, but might not be suitable for all environments.
-
-#### Stateful Deployment
-
- A stateful deployment of Prometheus should use persistent storage with [Persistent Volumes and Persistent Volume Claims][1] to maintain a correlation between a data volume and the Prometheus Pod.
- Persistent volumes can be static or dynamic and depends on the backend storage implementation utilized in environment in which the cluster is deployed. For more information, see the [Kubernetes documentation on types of persistent volumes][2].
-
-#### Quick start
+The instructions above show how to access the Prometheus and Grafana web interfaces using `kubectl port-forward`.
+Sample `HTTPProxy` resources in the `examples/` directory can also be used to access these through your Contour installation:
 
 ```sh
-# Deploy 
-$ kubectl apply -f examples/prometheus
+$ kubectl apply -f examples/prometheus/httpproxy.yaml
+$ kubectl apply -f examples/grafana/httpproxy.yaml
 ```
 
-#### Access the Prometheus web UI
+### Scrape Contour and Envoy metrics
+
+To enable Prometheus to scrape metrics from the Contour and Envoy pods, some small customizations are needed to the sample deployment:
+
+Update `ClusterRole` to enable the installed Prometheus instance to monitor `Pods` and other resources in all namespaces:
 
 ```sh
-$ kubectl -n projectcontour-monitoring port-forward $(kubectl -n projectcontour-monitoring get pods -l app=prometheus -l component=server -o jsonpath='{.items[0].metadata.name}') 9090:9090
+$ kubectl patch clusterrole prometheus-k8s --type=json --patch-file examples/prometheus/clusterrole-patch.json
 ```
 
-then go to `http://localhost:9090` in your browser.
-
-#### Access the Alertmanager web UI
+Now add [`PodMonitor`][6] resources for scraping metrics from Contour and Envoy pods in the `projectcontour` namespace:
```

Comment:
We use `PodMonitor` internally because when the `pods` of Envoy become not-ready during shutdown (preStop) you end up losing prom scrapes if you use `ServiceMonitor`